exercises/clock/clock.exs

```diff
@@ -0,0 +1,28 @@
+defmodule Clock do
+  defstruct hour: 0, minute: 0
+
+  @doc """
+  Returns a string representation of a clock:
+
+      iex> Clock.new(8, 9) |> to_string
+      "08:09"
+  """
+  @spec new(integer, integer) :: Clock
+  def new(hour, minute) do
+  end
+
+  @doc """
+  Adds two clock times:
+
+      iex> Clock.add(10, 0) |> Clock.add(3) |> to_string
+      "10:03"
+  """
+  @spec add(Clock, integer) :: Clock
+  def add(%Clock{hour: hour, minute: minute}, add_minute) do
+  end
+
+  defimpl String.Chars, for: Clock do
+    def to_string(%Clock{hour: hour, minute: minute}) do
```

Comment:
@parkerl `Protocol` does not force to use a `struct`. One can always implement the protocol on some other built-in data type (which I don't think is a good practice in this case but can be done). 
Here is an example of using `tuple` instead of `struct` keeping the public API exactly the same: https://gist.github.com/tejasbubane/9e7ae3253756edf093dbc98b4e8e65ff
All the test-cases pass without any change.